File: LinkedList-JavaScript/4-1.js
/*
Linked Lists
*/

class Node{
    /**
     * constructor(Node maker)
     * @data => yangi tugun kiriting.
     */
    constructor(data){
        this.data = data,
        this.next = null;
    }
}

/**
 * LinkedList Class
 */
class Linked_list{
    constructor() {
        this.head = null;
    }

    /**
     * console ga natijalarni chiqarish
     */
    printList() {
        let temp = this.head
        while (temp){
            console.log(temp['data'])
            temp = temp.next;
        }
    }

}


//  Linked list obyekt yaratib olamiz
const Llist = new Linked_list();



// Uchta node (tugun) yaratamiz
Llist.head = new Node("Dushanba")
const tuesday = new Node("Seshanba")
const wednesday = new Node("Chorshanba")





// Tugunlarni bog'laymiz
Llist.head.next = tuesday
tuesday.next = wednesday





// Linked Listni konsolga chiqaramiz
Llist.printList()

File: 3-LinkedList-Python/4-1.py
"""
Linked lists
"""

class Node:
    """Tugun (node) obyekti, ya'ni boshqa ma'lumotgan yo'naltirib turadi"""
    def __init__(self,data):
        self.data = data
        self.next = None

class LinkedList:
    """Linked List obyekti"""
    def __init__(self):
        self.head = None


### Linked list obyekt yozamiz
Llist = LinkedList();


### Uchta node (tugun) yaratamiz
Llist.head = Node("Dushanba")
tuesday = Node("Seshanba")
wednesday = Node("Chorshanba")



print(Llist.head.data) # Dushanba
print(Llist.head.next) # None, tugunlarni bog'lashimiz kerak


print(tuesday.data) # Seshanba
print(wednesday.data+'\n') # Chorshanba





### Tugunlarni bog'laymiz
Llist.head.next = tuesday
tuesday.next = wednesday

print(Llist.head.data) # Dushanba
print(Llist.head.next.data) # None ning o'rniga obyekt chiqishi kerak, Seshanba
print(Llist.head.next.next.data) # Chorshanba
